pkg/planner/core/exhaust_physical_plans.go

```diff
@@ -2714,10 +2721,29 @@ func (p *LogicalExpand) ExhaustPhysicalPlans(prop *property.PhysicalProperty) ([
 			ExtraGroupingColNames: p.ExtraGroupingColNames,
 		}.Init(p.SCtx(), p.StatsInfo().ScaleByExpectCnt(prop.ExpectedCnt), p.QueryBlockOffset(), mppProp)
 		expand.SetSchema(p.Schema())
-		return []base.PhysicalPlan{expand}, true, nil
+		physicalExpands = append(physicalExpands, expand)
+		// when the MppTaskType is required, we can return the physical plan directly.
+		if prop.TaskTp == property.MppTaskType {
+			return physicalExpands, true, nil
+		}
 	}
-	// if MPP switch is shutdown, nothing can be generated.
-	return nil, true, nil
+	// for property.RootTaskType, we can give a TiDB Expand.
+	{
+		taskTypes := []property.TaskType{property.CopSingleReadTaskType, property.CopMultiReadTaskType, property.RootTaskType}
```

Comment:
Do we need to keep all taskTypes in here? Because there is only expand operator in tidb, so why we still need to keep copsingle and copmulti?

Reply:
when we come here, the prop asks for a root expand execution, while for its children prop, there is no limitation here.
```
Expand(Root)
 + IndexReader( from single cop prop)
     + IndexRangeScan

Expand(Root)
 + IndexLookupReader (from multi cop prop)
     + IndexRangeScan
     + TiDBRowIDScan

Expand(Root)
 + OtherRootOp(Root) (from root prop)
     + OtherOps 

Expand(Root)
 + OtherMppOps (Mpp) (from mpp prop)
     + OtherOps 
```
the four cases here are all reasonable